app/src/main/java/com/willowtree/vocable/presets/adapter/AddToCategoryPickerViewModel.kt

```diff
@@ -47,12 +49,22 @@ class AddToCategoryPickerViewModel : BaseViewModel() {
         }
     }
 
-    fun buildCategoryList(phraseString: String) {
-        // map category to whether or not the phrase is in that category
+    fun getCategoryList() {
+        backgroundScope.launch {
+            liveCategoryList.postValue(presetsRepository.getUserGeneratedCategories())
+        }
+    }
+
+    fun buildCategoryList(phraseString: String, categories: List<Category>): Map<Category, Boolean> {
         val phraseMap = mutableMapOf<Category, Boolean>()
 
-        backgroundScope.launch {
-            userCategories = presetsRepository.getUserGeneratedCategories()
+        categories.forEach {
+            phraseMap[it] = false
+        }
+
+        return runBlocking {
```

Comment:
Why the switch to `runBlocking` here? This will block the UI thread for the user.